test/fixtures/build-default/src/foo.ts

```diff
@@ -1 +1 @@
-export const foo = () => 'bar';
+export const foo = (foo?: { bar: string }) => foo?.bar || 'bar';
```

Comment:
If a test is going to be added for this, I think it should be a separate file named appropriately, instead of changing an existing test.

Also this only tests optional chaining, but nullish coalescing was also added here. If there will be a test for this, then it should test both (also PR name should probably be changed to note it also supports nullish coalescing usage)

Reply:
I added a `syntax` directory which contains a specific file for each syntax features. Thanks